### DISCUSSION TITLE
Ajuda para aprender a programar

### DISCUSSION THREAD

**Discussion by AntonioDeveloper** [frontendbr/forum#1451] [Discussão]
Quero ficar mais confortável em programação, em especial JS.
Há tempos, eu procuro um meio legal de fazer isso. Mas ainda não encontrei uma forma eficiente de praticar.
Pois, atualmente, o que conheço melhor é HTML, CSS. Até entendo lógica de programação básica e alguns comandos de JS mas, se me derem um problema pra resolver, eu travo.
Na teoria dos cursos que faço, tudo bem, na hora de praticar complica.
Seria bom, se houvesse um jeito de eu praticar e ver como estou me saindo.
Por exemplo, o Code Academy tem desafios e uma plataforma onde eu posso programar. E lá, o sistema me diz o que estou fazendo errado.
O mau dele é ser cobrado em dólar.
Alguém tem alguma dica pra me dar, de como contornar esse problema?
Obrigado 😊

**Comment by gabriel-hahn** [comment]
Olá Antonio,

Se teu foco é realmente só a lógica de programação utilizando o JavaScript, existem alguns sites como o [HackerRank](https://www.hackerrank.com/) que te propõe desafios.
Para aprender mais sobre os métodos e funcionamento do JS, recomendo o [FreeCodeCamp](https://www.freecodecamp.org/) e os cursos gratuitos da [Rocketseat](https://rocketseat.com.br/) (Português). Todos esses são gratuitos.
Existe também os vídeos do Rodrigo Branas no Youtube, são bem legais e fáceis de entender.

Tenta criar projetos simples e ir avançando aos poucos. Eu também faço bastante cursos e na hora da prática tenho bastante problemas, porém é algo normal e tu vai aprendendo e absorvendo esse conhecimento conforme vai treinando.

**Comment by salatielsql** [comment]
> Para aprender mais sobre os métodos e funcionamento do JS, recomendo o [FreeCodeCamp](https://www.freecodecamp.org/)

Eu também aprendi muito de JS com o FreeCodeCamp. Super recomendo!

**Comment by mdandrade** [comment]
Cara, tenta a sorte de mandar um e-mail pra eles darem desconto. Já consegui algumas vezes explicando que vivo em país onde o câmbio afeta muito e eles contornam, kk

___________________________________
Atenciosamente,

Márcio Daniel Andrade

________________________________

**Comment by AntonioDeveloper** [comment]
Eu agradeço a todos os que se dispuseram a me ajudar! Vou experimentar tudo o que vocês me indicaram. Quero muito dominar programação. 👊

**Comment by fernandoRj** [comment]
vai no w3school veja como se faz coisas basicas no js, cola no seu editor, e depure no navegador, leia o que o codigo faz, escreve os processos que o codigo realiza em um bloco de notas e depois tente codar o que tu escreveu,

**Comment by henrique-barbase** [comment]
Recomendo muito o canal do [Rodrigo Branas](https://www.youtube.com/user/rodrigobranas/videos)

**Comment by AntonioDeveloper** [comment]
@henrique-barbase, muito obrigado pela dica! O Rodrigo Brancas acabou de ganhar mais um inscrito!
